feat(medusa,dashboard): Add support for configurable additional columns in entity views

docloulou opened the pull request:
This pull request enhances the entity column generation logic in the admin views by adding support for including fields from additional GraphQL types, specifically for the `orders` entity. The changes allow more flexible and comprehensive column definitions by pulling in fields from related types (like `OrderDetail`) and updating the filtering and type resolution logic accordingly.

**This adding payment_status & fulfillment_status from OrderDetail in view configuration feature**

**Entity column generation enhancements:**

* Added the `ADDITIONAL_ENTITY_TYPES` mapping to specify extra GraphQL types whose fields should be included for certain entities (currently, `OrderDetail` for `orders`). ([packages/medusa/src/api/admin/views/[entity]/columns/helpers.tsR260-R263](diffhunk://#diff-ce197feb4e4d1273d9ee19126e284b65fdb4367f0871774a75add5c8cd749d02R260-R263))
* Updated the column generation process to collect fields from both the main entity type and any additional types, while properly filtering out arrays and excluded fields. ([packages/medusa/src/api/admin/views/[entity]/columns/helpers.tsR340-R345](diffhunk://#diff-ce197feb4e4d1273d9ee19126e284b65fdb4367f0871774a75add5c8cd749d02R340-R345), [packages/medusa/src/api/admin/views/[entity]/columns/helpers.tsR373-R414](diffhunk://#diff-ce197feb4e4d1273d9ee19126e284b65fdb4367f0871774a75add5c8cd749d02R373-R414))
* Changed field lookup logic to use a unified `entityFields` object and a new `additionalFieldDefinitions` map for extra fields, ensuring correct type info resolution for all columns. ([packages/medusa/src/api/admin/views/[entity]/columns/helpers.tsL344-R354](diffhunk://#diff-ce197feb4e4d1273d9ee19126e284b65fdb4367f0871774a75add5c8cd749d02L344-R354), [packages/medusa/src/api/admin/views/[entity]/columns/helpers.tsL411-R463](diffhunk://#diff-ce197feb4e4d1273d9ee19126e284b65fdb4367f0871774a75add5c8cd749d02L411-R463))

docloulou committed:
feat: Enhance entity column generation with additional types support

- Introduced `ADDITIONAL_ENTITY_TYPES` to allow for the inclusion of extra fields from specified entity types.
- Updated `generateEntityColumns` function to incorporate additional fields while filtering out problematic entries.
- Improved handling of field definitions to ensure consistency and correctness in the generated columns.

docloulou committed:
changeset

willbouch commented:
Haven't had time to take a deep look. I haven't forgotten you. Will do when I have a sec

docloulou commented:
> Haven't had time to take a deep look. I haven't forgotten you. Will do when I have a sec

Thanks,

its works as is, only i18n traduction is missing/misslinked 
<img width="1332" height="770" alt="image" src="https://github.com/user-attachments/assets/e7406877-e68c-4312-8d03-26da68476820" />

willbouch commented on .changeset/silly-feet-sniff.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@medusajs/medusa": minor
```
Hey! Your change is non-breaking correct? It should be`patch` here if that is the case

willbouch commented on .changeset/silly-feet-sniff.md:
```diff
@@ -0,0 +1,5 @@
+"@medusajs/medusa": minor
+---
+
+Enhance entity column generation with additional types support
```
Also, nitpicky but we try to have the PR name here. You have it as `fix(): Fix columns helper (view entities columns)` for now. We usually try to put the packages changed in the (). In your case, you would have `fix(medusa): fix columns helper (view entities columns)`

willbouch reviewed (commented):
I tested it out locally and it works fine! What do you suggest for translations though? 

Also, what should we do about filters? It is nice to see many more columns, but the filters do not follow. Maybe there is a way to dynamically add new ones? 🤔 It wasn't there already, but maybe we could find a way to "fix"

docloulou committed:
- Enhanced translation logic for payment and fulfillment statuses to support dynamic key generation.

docloulou committed:
changetset update

docloulou committed:
changeset update 2

docloulou commented:
Here you go, I’ve added the translation — it’s a bit tricky and not flexible at all for custom modules… (unless we use already-translated fields).

As for the filters, if I have time and make enough progress on my project, I’ll open another PR ;)

docloulou commented:
> I tested it out locally and it works fine! What do you suggest for translations though? 
> 
> 
> 
> Also, what should we do about filters? It is nice to see many more columns, but the filters do not follow. Maybe there is a way to dynamically add new ones? 🤔 It wasn't there already, but maybe we could find a way to "fix"

I thought about filters. Even if we add filters dynamically, the API route middleware restricts certain fields. This part should also be reworked, or gates should only be opened for admins routes.
Have you already had internal discussions about that?

willbouch commented:
> Here you go, I’ve added the translation — it’s a bit tricky and not flexible at all for custom modules… (unless we use already-translated fields).
> 
> As for the filters, if I have time and make enough progress on my project, I’ll open another PR ;)

Did you push? haha

docloulou commented:
> > Here you go, I’ve added the translation — it’s a bit tricky and not flexible at all for custom modules… (unless we use already-translated fields).
> > As for the filters, if I have time and make enough progress on my project, I’ll open another PR ;)
> 
> Did you push? haha

Yes, change were made in packages/admin/dashboard/src/lib/table/cell-renderers.tsx (see commit) translation were already there but pointed to status.xxxx instead of order.payment/fulfillment.status.xxxx  (i just used the context from the column api)

willbouch commented on packages/admin/dashboard/src/lib/table/cell-renderers.tsx:
```diff
@@ -25,6 +25,14 @@ export type RendererRegistry = Map<string, CellRenderer>
 
 const cellRenderers: RendererRegistry = new Map()
 
+const toCamelCase = (str: string): string => {
```
It would be nice to move that in a helper file. There are many already in dashboard/src/lib

docloulou replied:
you mean a translation-helper.ts or something with just one func toCamelCase  ?

willbouch replied:
Maybe in `common.ts` ?

docloulou committed:
- Introduced `toCamelCase` function in `common.ts` for converting strings to camel case format.
- Removed the local implementation of `toCamelCase` from `cell-renderers.tsx` and imported the new utility function for better code reuse.

docloulou commented:
@willbouch done, hope this will included in the next release :)

willbouch commented:
Will try to get this merged tomorrow

willbouch committed:
Merge branch 'develop' into fix-columns-helper

willbouch commented:
@docloulou I have the green flag to merge this. You seem to change the admin, so we should update the changeset to reflect that please :) Also a patch.

Also, this looks more like a feature than a fix don't you think? Maybe the PR name + the message in the changeset should be adapted to write it more like a feat?

docloulou committed:
Update silly-feet-sniff.md

docloulou commented:
> @docloulou I have the green flag to merge this. You seem to change the admin, so we should update the changeset to reflect that please :) Also a patch.
> 
> 
> 
> Also, this looks more like a feature than a fix don't you think? Maybe the PR name + the message in the changeset should be adapted to write it more like a feat?

Yes you right, I changed the PR title and the changeset.
Tell me if everything is alright.

willbouch committed:
Merge branch 'develop' into fix-columns-helper
